src/staticarrays_support.jl

```diff
@@ -26,4 +26,9 @@ StructArrays.component(s::StaticArray, i) = getindex(s, i)
     invoke(StructArrays.staticschema, Tuple{Type{<:Any}}, T)
 end
 StructArrays.component(s::FieldArray, i) = invoke(StructArrays.component, Tuple{Any, Any}, s, i)
-StructArrays.createinstance(T::Type{<:FieldArray}, args...) = invoke(createinstance, Tuple{Type{<:Any}, Vararg}, T, args...)
\ No newline at end of file
+StructArrays.createinstance(T::Type{<:FieldArray}, args...) = invoke(createinstance, Tuple{Type{<:Any}, Vararg}, T, args...)
+
+function Base.copy(bc::Broadcasted{StructArrayStyle{StaticArrayStyle{N},N}}) where {N}
+    B = convert(Broadcasted{StructArrayStyle{Broadcast.DefaultArrayStyle{N},N}}, bc)
+    copy(B)
```

Comment:
```suggestion
    bc′ = convert(Broadcasted{StructArrayStyle{Broadcast.DefaultArrayStyle{N},N}}, bc)
    return copy(bc′)
```
Just to be consistent with the notation below.

It may also be useful to add a comment as to what exactly is happening with the conversion above. I'm also a bit puzzled by `DefaultArrayStyle` here. Are we ending up calling the `similar` method defined below? In that case, are we allocating regular `Array`s? That seems like something that should be avoided.

Reply:
The best choise for me is converting `bc` to `Broadcasted{StaticArrayStyle{N}}` and return a `SArray`.
But `StaticArrays.jl` hardcoded some type check in their generated `copy`: 
https://github.com/JuliaArrays/StaticArrays.jl/blob/81f6c7c32db3cf6901d4290a7f6679886f03ef23/src/broadcast.jl#L100-L102

So we have to return a struct-`SizedArray` instead.